### PR TITLE
Sodaracers Python Encoding (walk_creator & hand-designed Square, Radial, and CPPN seeds)

### PR DESCRIPTION
**Pull request by mathyouf**
## Done:
- [x] The `walk_creator` has the `add_joint`, `add_muscle`, and `get_walker` methods
-  [x] `get_walker` serializes to the desired intermediate Sodaracer representation when called

- [x] `Walker.validate` ensures the Sodaracer will not break the underlying Box2D physics engine
- [x] `square.py`, `radial.py`, `CPPN_fixed.py`, and `CPPN_mutable.py` are instantiations of hand-designed Sodaracers, which the diff model can be applied to (copied from pg. 46-48)
- [x] `README.md` added with references to the paper

## To Do:
- [ ] (Optional) Check validity of serialized Seeds against Sodaracer Environment: [dmahan93/pyIesorPhysics  (WiP)](https://github.com/dmahan93/pyIesorPhysics)

### CODE REVIEW

File: walker/walk_creator.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    def __eq__(self, other):
+        # Check if other is a Dictionary
+        if isinstance(other, dict):
+            return self.serialize_walker() == other
+        return self == other
+
+    def validate(self):
+        """logic for ensuring that the Sodaracer will not break the underlying Box2D physics engine
+            a) that each joint is connected only to so many muscles
```
Comment (mathyouf) [inline walker/walk_creator.py:55]: How many muscles

File: walker/walk_creator.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            b) that the strength of muscles is limited
```
Comment (mathyouf) [inline walker/walk_creator.py:56]: What should be the max strength?

File: walker/walk_creator.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            c) that there is a minimum distance between joints
```
Comment (mathyouf) [inline walker/walk_creator.py:57]: What should be the minimum distance

File: walker/walk_creator.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        return str(self.serialize_walker())
+
+    def __eq__(self, other):
+        # Check if other is a Dictionary
+        if isinstance(other, dict):
+            return self.serialize_walker() == other
+        return self == other
+
+    def validate(self):
+        """logic for ensuring that the Sodaracer will not break the underlying Box2D physics engine
```
Comment (mathyouf) [inline walker/walk_creator.py:54]: Any other engine breaking logic checks we should include not mentioned below? (below are the three examples from the paper).

File: walker/walk_creator.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+            a) that each joint is connected only to so many muscles
+            b) that the strength of muscles is limited
+            c) that there is a minimum distance between joints
```
Comment (mathyouf) [inline walker/walk_creator.py:57]: Any other engine breaking logic checks we should include not mentioned above? 
(these are the three examples from the paper).

File: walker/walk_creator.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        return j
+
+    def add_muscle(self, j0, j1, *args):
+        """add a point mass"""
+        m = Muscle(j0, j1, args)
+        self.muscles.append(m)
+        return m
+
+    def get_walker(self):
+        """Python dictionary with keys such as “joints” and “muscles”"""
```
Comment (honglu2875) [inline walker/walk_creator.py:101]: The quotation marks are non-ascii characters. I was testing the code on Python 3.7 and got an error about those. 
Maybe change to this '...' ?

Comment (mathyouf) [reply walker/walk_creator.py]: Fixed! Nice catch, it was a copy-paste from the ELM paper.

**Review by herbiebradley — APPROVED** [review]
Thanks, great work!

**Pull request merged**